### PR TITLE
feat(specs): convert nested specs to dictionnary of dm_env specs

### PR DESCRIPTION
**Pull request by clement-bonnet**
- fix: remove pygame legacy
- feat: implement nested dm_env specs
- feat: improve specs tests

### CODE REVIEW

File: jumanji/specs_test.py [hunk 1]
```diff
@@ -15,202 +15,162 @@
...
 import numpy as np
 import pytest
 from chex import assert_trees_all_equal
 
 from jumanji import specs
+from jumanji.testing.pytrees import assert_tree_with_leaves_of_type
 
 
-class ThirdClass(NamedTuple):
+class SinglyNested(NamedTuple):
```
Comment (djbyrne) [inline jumanji/specs_test.py:33]: I would probably go with `SingleNestedNamedTuple` I feel it makes it clearer on what it is

File: jumanji/specs_test.py [hunk 2]
```diff
@@ -15,202 +15,162 @@
...
+    array: chex.Array
     bounded_array: chex.Array
     multi_discrete_array: chex.Array
 
 
-class SecondClass(NamedTuple):
-    array: chex.Array
-    third_class: ThirdClass
+class DoublyNested(NamedTuple):
```
Comment (djbyrne) [inline jumanji/specs_test.py:39]: Same with the following, DoubleNestedNamedTupel and TripleNestedNamedTuple

File: jumanji/specs_test.py [hunk 3]
```diff
@@ -648,7 +612,9 @@ def test_bounded_array(self) -> None:
     def test_discrete_array(self) -> None:
```
Comment (djbyrne) [inline jumanji/specs_test.py:612]: NIT: some tests have docstring descriptions and some don't. I'll leave it up to you if they are all necessary

File: jumanji/specs_test.py [hunk 4]
```diff
@@ -15,202 +15,162 @@
...
-    not Jumanji specs.
-    """
-
-    def __init__(self) -> None:
-        super().__init__()
-        self.nested_spec = SinglyNestedSpec()
-        self.array_spec = specs.Array(self._shape, self._dtype)
-        self.not_spec_1 = [1.0]
-        self.not_spec_2 = (2.0,)
+class TestNestedSpec:
```
[the remark this answers is not in the document]
Comment (clement-bonnet) [reply jumanji/specs_test.py:108]: Good to know, thanks! We have used `TestXXX` classes throughout Jumanji to group tests that go together (and may share some class attributes), so doing this here as well for consistency, but you're saying this is actually not needed?

File: jumanji/specs_test.py (hunk 3 above)
Comment (clement-bonnet) [reply jumanji/specs_test.py:612]: Overall, I try to improve as I update the code. If I see tests that are unclear and do not have a docstring I will add one.

File: jumanji/specs.py [hunk 5]
```diff
@@ -587,11 +590,11 @@ def jumanji_specs_to_gym_spaces(
             dtype=spec.dtype,
             seed=None,
         )
-    # Nested spec such as EMSSpec, ItemSpec, and ObservationSpec in BinPack environment
     else:
+        # Nested spec.
         return gym.spaces.Dict(
             {
-                # Iterate over specs
+                # Iterate over specs.
```
[the remark this answers is not in the document]
Comment (clement-bonnet) [reply jumanji/specs.py:597]:
```suggestion
                # Iterate over specs
```

File: jumanji/specs.py [hunk 6]
```diff
@@ -540,11 +541,13 @@ def jumanji_specs_to_dm_env_specs(
...
         )
     else:
-        raise ValueError(
-            f"spec {spec} of type {type(spec)} is not available in a deepmind environment. "
-            "Please override the observation_spec or action_spec method to output spec of type "
-            "`dm_env.specs.Array`."
-        )
+        # Nested spec.
+        return {
+            # Iterate over specs.
```
Comment (clement-bonnet) [inline jumanji/specs.py:546]:
```suggestion
        # Nested spec
        return {
            # Iterate over specs
```

**Pull request merged**